src/integration/java/io/pinecone/integration/dataPlane/UpdateFetchAndQueryPodTest.java

```diff
@@ -248,79 +239,87 @@ public void updateAllParamsFetchAndQueryFutureTest() throws InterruptedException
                     true,
                     true).get();
 
-            ScoredVectorWithUnsignedIndices scoredVectorV1 = queryResponse.getMatches(0);
+            List<ScoredVectorWithUnsignedIndices> queryResults = queryResponse.getMatchesList();
+            ScoredVectorWithUnsignedIndices scoredVectorV1 = null;
+
+            for (ScoredVectorWithUnsignedIndices vector : queryResults) {
+                if (idToUpdate.equals(vector.getId())) {
+                    scoredVectorV1 = vector;
+                }
+            }
```

Comment:
The way this was being done originally confused me, and also seemed prone to possible flakiness, so I updated these in several test files.